packages/core-data/src/batch-processor.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+export default async function batchProcessor( requests ) {
+	const batchResponse = await apiFetch( {
+		path: '/v1/batch',
+		method: 'POST',
+		data: {
+			validation: 'require-all-validate',
+			requests: requests.map( ( request ) => ( {
+				path: request.path,
+				body: request.data, // Rename 'data' to 'body'.
+				method: request.method,
+				headers: request.headers,
+			} ) ),
+		},
+	} );
+
+	if ( batchResponse.failed ) {
+		return batchResponse.responses.map( ( response ) => ( {
+			error: response?.body,
+		} ) );
+	}
+
+	return batchResponse.responses.map( ( response ) => {
```

Comment:
It'd be nice if those also checked against `request.parse` and when `false` created the full response object.

Reply:
Sorry not sure what you mean here?